onnxruntime/contrib_ops/cuda/bert/bert_padding.cu

```diff
@@ -361,6 +362,46 @@ void LaunchTrtSequenceOffset(int* trt_mha_padding_offset,
   }
 }
 
+// Write is not effective
+__global__ void __launch_bounds__(kMAX_THREADS_PER_BLOCK)
+    getTrtSequenceOffset2d(int* trt_mha_padding_offset,
+                           const int* attention_masks,
+                           const int batch_size,
+                           const int sequence_length) {
+
+    typedef cub::BlockReduce<int, kMAX_THREADS_PER_BLOCK> BlockReduce;
+    __shared__ typename BlockReduce::TempStorage temp_storage;
+
+    const int batch_id = blockIdx.x;
+    const int* batch_mask = attention_masks + (batch_id * sequence_length);
+    int first_non_padding = sequence_length;
+    for (int i = threadIdx.x; i < sequence_length; i += blockDim.x) {
+      if (batch_mask[i]) {
```

Comment:
Need to handle the following mask
[[1,1, 1, 0, 0, 0],  // right side padding
 [0, 0, 0, 1, 1, 1], // left side padding
 [1, 1, 1, 1, 1, 1]] // no padding.
 
 Need a way to tell that mask is not supported by fused kernel like
[[1, 1, 0, 0, 1, 1]]

Reply:
support both left and right, reasonable requirements.